ppafm/ocl/field.py

```diff
@@ -292,25 +356,21 @@ def add_mult(self, array, scale=1.0, in_place=True, local_size=(32,), queue=None
         Returns:
             grid_out: Same type as self. New data grid with result.
         """
+
         array_in1 = self.cl_array
         array_in2 = array.cl_array
-        queue = queue or oclu.queue
-        if in_place:
-            grid_out = self
-            array_out = array_in1
-            self._array = None  # The current numpy array will be wrong after operation so reset it
-        else:
-            array_out = cl.Buffer(self.ctx, cl.mem_flags.READ_WRITE, size=array_in1.size)
-            array_type = type(self)  # This way so inherited classes return their own class type
-            grid_out = array_type(array_out, lvec=self.lvec, shape=self.shape, ctx=self.ctx)
+        grid_out = self._prepare_same_size_output_grid(array_in1, in_place)
```

Comment:
Are you sure, that removing the queue from here and adding it after preparation of grid out is safe?

Reply:
Should be safe. The array preparation does not use the variable.